### DISCUSSION TITLE
Add support for speech recognition

### DISCUSSION THREAD

**Discussion by mrosso10** [d99kris/nchat#268] [Ideas]
Hi, I'm currently using [https://github.com/openai/whisper](https://github.com/openai/whisper) to convert audio to text, and it would be nice to have it integrated to nchat.
I'm currently experimenting with the nchat code and I think I could start a PR with a "minimum viable feature", that would consist of:

- A shortcut that (having an audio message selected) spawns an external (configurable) process
- That process (whisper) would transcribe the audio and save the output to the same folder the audio is in (~/.config/nchat/profiles/...), with .txt extension
- A change to uihistoryview.cpp for rendering the audio transcription near the attachment file name.

Other improvements that would be nice:

- An sign near the attachment filename that indicates if the file is being processed
- Saving the transcription in the chat export

**Comment by d99kris** [comment]
Hi @mrosso10 - it's a cool idea, but at the same time I'm quite hesitent to adding corner-case features that increase maintainence work (for the core functionality). I suppose we can see how well encapsulated the functionality becomes and then might can consider merging such a PR, but I cannot promise it.

**Comment by Bellavene** [comment]
Yes, that would be great. While I don't like audio messages at all. Some people send them anyway, and it is a struggle if it has important information, would be great to have a possibility to read it inside nchat.

**Reply by mrosso10** [reply]
Hi @Bellavene, I'm currently using this patch: https://github.com/mrosso10/nchat/commit/7b03de5b7c92c4dd450a7271c8d89091264ed09d

It's not a clean code at all because it was just a test for personal use, but it works.

If you could manage to install a speech recognition software (I'm using [whisper](https://github.com/openai/whisper)*)
and want me to help you with that commit, please let me know.

\* Warning: whisper eats a lot of memory (~ 4 GB)

**Reply by Bellavene** [reply]
Hello @mrosso10,
Thank You, I will check it out. Interesting, is there a way to use the system's built-in speech recognition?

**Reply by mrosso10** [reply]
Do you mean the whatsapp's feature? I don't know, but I doubt it given it's a very new feature.

**Reply by Bellavene** [reply]
Loosing 4GB of ram for that feature is a bit overkill. I meant the built-in speech recognition in macos.
